src/back/msl/writer.rs

```diff
@@ -829,9 +842,23 @@ impl<W: Write> Writer<W> {
                     if !is_scoped {
```

Comment:
would it make sense to do `else if op == crate::BinaryOperator::Mul` and put the logic there instead?